src/pc/weapon_pc.c

```diff
@@ -0,0 +1,863 @@
+        GetWeaponIdDummy,
+        LoadWeaponPaletteDummy,
+        EntityWeaponShieldSpellDummy,
+        func_ptr_80170024Dummy,
+        func_ptr_80170028Dummy,
+    },
+    {
+        EntityWeaponAttackDummy,
+        func_ptr_80170004Dummy,
+        func_ptr_80170008Dummy,
+        func_ptr_8017000CDummy,
+        func_ptr_80170010Dummy,
+        func_ptr_80170014Dummy,
+        GetWeaponIdDummy,
+        LoadWeaponPaletteDummy,
+        EntityWeaponShieldSpellDummy,
+        func_ptr_80170024Dummy,
+        func_ptr_80170028Dummy,
+    },
+    {
+        EntityWeaponAttackDummy,
+        func_ptr_80170004Dummy,
+        func_ptr_80170008Dummy,
+        func_ptr_8017000CDummy,
+        func_ptr_80170010Dummy,
+        func_ptr_80170014Dummy,
+        GetWeaponIdDummy,
+        LoadWeaponPaletteDummy,
+        EntityWeaponShieldSpellDummy,
+        func_ptr_80170024Dummy,
+        func_ptr_80170028Dummy,
+    },
+    {
+        EntityWeaponAttackDummy,
+        func_ptr_80170004Dummy,
+        func_ptr_80170008Dummy,
+        func_ptr_8017000CDummy,
+        func_ptr_80170010Dummy,
+        func_ptr_80170014Dummy,
+        GetWeaponIdDummy,
+        LoadWeaponPaletteDummy,
+        EntityWeaponShieldSpellDummy,
+        func_ptr_80170024Dummy,
+        func_ptr_80170028Dummy,
+    },
+    {
+        EntityWeaponAttackDummy,
+        func_ptr_80170004Dummy,
+        func_ptr_80170008Dummy,
+        func_ptr_8017000CDummy,
+        func_ptr_80170010Dummy,
+        func_ptr_80170014Dummy,
+        GetWeaponIdDummy,
+        LoadWeaponPaletteDummy,
+        EntityWeaponShieldSpellDummy,
+        func_ptr_80170024Dummy,
+        func_ptr_80170028Dummy,
+    },
+    {
+        EntityWeaponAttackDummy,
+        func_ptr_80170004Dummy,
+        func_ptr_80170008Dummy,
+        func_ptr_8017000CDummy,
+        func_ptr_80170010Dummy,
+        func_ptr_80170014Dummy,
+        GetWeaponIdDummy,
+        LoadWeaponPaletteDummy,
+        EntityWeaponShieldSpellDummy,
+        func_ptr_80170024Dummy,
+        func_ptr_80170028Dummy,
+    },
+    {
+        EntityWeaponAttackDummy,
+        func_ptr_80170004Dummy,
+        func_ptr_80170008Dummy,
+        func_ptr_8017000CDummy,
+        func_ptr_80170010Dummy,
+        func_ptr_80170014Dummy,
+        GetWeaponIdDummy,
+        LoadWeaponPaletteDummy,
+        EntityWeaponShieldSpellDummy,
+        func_ptr_80170024Dummy,
+        func_ptr_80170028Dummy,
+    }};
+
+SpriteParts* g_Animset[1];
+
+int readSubsetToBuf(char* filename, char* dest, size_t start, size_t size) {
+    FILE* file = fopen(filename, "rb");
+
+    if (file == NULL) {
+        printf("Failed to open file");
+        return 1;
+    }
+
+    // seek to target position
+    fseek(file, start, SEEK_SET);
+
+    // Read the file into the buffer
+    size_t read_size = fread(dest, 1, size, file);
+    if (read_size != size) {
+        printf("Failed to read file");
+        fclose(file);
+        return 1;
+    }
```

Comment:
Failure here could be due to `EOF` or other error. Instead of `printf` do you think `perror` and/or an `feof` check to specify the failure would be appropriate?

Reply:
OK I added checks for ferror and feof